### ISSUE TITLE
Support for headless policies

### ISSUE THREAD

**Issue by aried3r** [chrisalley/pundit-matchers#4]
Are there any plans to add support for [headless policies](https://github.com/elabs/pundit#headless-policies)?

Currently, because of the references to `record`, this happens:

``` Ruby
NoMethodError: undefined method `record' for #<DashboardPolicy:0x007faee3454130>
```

**Comment by chrisalley** [comment]
Can you please paste or link to the code which is causing that error? I created a policy like this:

```ruby
class DashboardPolicy < Struct.new(:user, :record)
  def show?
    true
  end
end
```
With this corresponding spec:

```ruby
require 'rails_helper'

describe DashboardPolicy do
  subject { DashboardPolicy.new(user, :dashboard) }

  context "being a visitor" do
    let(:user) { nil }

    it { should permit_action(:show) }
  end
end
```

With this code, the test passed without error.

**Comment by aried3r** [comment]
Sorry for the late reply. It was a mistake on my side, we were still using custom matchers with the exact same name and that was causing the problems, but only with headless policies which is why I thought it might be this gem.

Everything works now, however the error messages are not as useful for headless policies:
```Ruby
class DashboardPolicy < Struct.new(:user, :dashboard)
  def index?
    user.admin?
  end
end
```

```Ruby
require 'rails_helper'

RSpec.describe DashboardPolicy do
  subject { described_class.new(user, :dashboard) }

  describe 'as an admin user' do
    let(:user) { FactoryGirl.create :user, :admin }

    it { is_expected.to forbid_action(:index) }
  end
end
```

Result:
```
NoMethodError: undefined method `record' for #<DashboardPolicy:0x007fab38748d68>
```

**Comment by chrisalley** [comment]
The error messages could certainly be improved. This will require some refactoring. I'll leave the issue open; if/when there's time I'll see what can be done to make the error messages more intuitive.

**Comment by anton-smagin** [comment]
I made a PR for fix this error, what do you think about it?
https://github.com/chrisalley/pundit-matchers/pull/12

**Comment by chrisalley** [comment]
Thanks. I've merged the PR and released it as version 1.3.1.
